includes/options/class-amp-options-rest-controller.php

```diff
@@ -0,0 +1,160 @@
+<?php
```

Comment:
Actually, it seems like all we need to do to expose these options on the REST API is to pass the `show_in_rest` argument when registering the `amp-options` setting:

https://github.com/ampproject/amp-wp/blob/07b638e1f931bbcdfbc9162940763da1cb301dd9/includes/options/class-amp-options-manager.php#L40-L47

That argument requires that a settings schema be supplied if the setting type is an `array`, which is true in our case.

With this approach, we automatically get permission checks as well as validation of the setting when it's updated (via `AMP_Options_Manager::validate_options`).

Reply:
Awesome idea. I'm going to give it a shot. Maybe cut down this diff by 160 lines 😄